backend/python-functions/data_ingest.py

```diff
@@ -0,0 +1,145 @@
+
+    download_location = f'/tmp/{file_name}'
+
+    blob.download_to_filename(download_location)    
+
+    return download_location
+
+def reshape_facilities_data(file_location):
+    """
+    This function reshapes the daily Covid case data provided in CSV format into a 
+    dictionary structure that: 1) Removes repetitive and unwanted data and 2) Aligns
+    with how we ultimately persist our documents within Firestore.
+
+    For example, the following CSV:
+
+    Date,Facility Type,State,Canonical Facility Name,Pop Tested,Pop Tested Positive, etc.
+    2020-06-01,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+    2020-06-02,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+    2020-06-03,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+    2020-06-04,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+
+    Would be reshaped to:
+
+    {
+      "Colorado::Denver Reception & Diagnostic Center": {
+        "canonicalName": "Denver Reception & Diagnostic Center",
+        "facilityType": "State Prisons",
+        "state": "Colorado",
+        "covidCases": {
+          "2020-06-01": {
+            "popDeaths": 0,
+            "popTested": 253,
+            "popTestedPositive": 2
+          },
+          "2020-06-02": {
+            "popDeaths": 0,
+            "popTested": 253,
+            "popTestedPositive": 2
+          },
+          "2020-06-03": {
+            "popDeaths": 0,
+            "popTested": 253,
+            "popTestedPositive": 2
+          },
+          "2020-06-04": {
+            "popDeaths": 0,
+            "popTested": 253,
+            "popTestedPositive": 2
+          }
+        }
+      },
+      ...
+    }
+    """
+    facilities = {}    
+    with open(file_location) as csv_file:
+        reader = csv.DictReader(csv_file, delimiter=',')
+
+        for row in reader:
+            state = row["State"].strip()
+            canonical_facility_name = row["Canonical Facility Name"].strip()
+            facility_type = row['Facility Type'].strip()
+            key = f'{state}::{canonical_facility_name}'
+            
+            if(key in facilities):
+                # If the facility already exists, append the case counts for a given day.
+                facilities[key]['covidCases'][row['Date']] = build_covid_case_counts(row)
+            else:
+                # If the facility does not already exist, save its metadata along with
+                # the case counts for a given day.
+                facilities[key] = {
+                    'canonicalName': canonical_facility_name,
+                    'facilityType': facility_type,
+                    'state': state,
+                    'covidCases': {
+                        f'{row["Date"]}': build_covid_case_counts(row)
+                    }
+                }
+
+    return facilities
+
+def persist(facilities):
```

Comment:
For now, the way this works is that the entire Covid case data CSV file will be reshaped and iterated through in its entirety each day.  We are not doing a daily diff / append.  The significance of this is that if we get a file that changes its case counts for a facility for a given day in the past, those case counts in the past will be updated.  I believe this should actually be the desired functionality.  Our Firestore data will reflect the most recent set of truth as we know it.

It may sound like this would cause a lot of unnecessary database updates, but Firestore seems to be smart enough to not write or update documents unless they are new or have changed.